packages/core/src/mpi/get-patient-by-demo.ts

```diff
@@ -0,0 +1,60 @@
+import { Patient, PatientData } from "../domain/medical/patient";
+import { PatientLoader } from "../domain/medical/patient-loader";
+import {
+  jaroWinklerSimilarity,
+  matchingContactDetailsRule,
+  matchingPersonalIdentifiersRule,
+  matchPatients,
+} from "./match-patients";
+import { mergeWithFirstPatient } from "./merge-patients";
+import { normalizePatient } from "./normalize-patient";
+import { patientToPatientMPI } from "./shared";
+
+const SIMILARITY_THRESHOLD = 0.96;
+
+/**
+ * Retrieves a patient based on their demographic information. Utilizes functions
+ * from the MPI: normalization, finding(blocking), matching, merging
+ *
+ * @param cxId - The ID of the patient in the external system.
+ * @param demo - The demographic information of the patient.
+ * @returns The matched patient object if found, otherwise undefined.
+ */
+export const getPatientByDemo = async ({
```

Comment:
This is one of the functions that would be good to have a few unit tests.

Reply:
Thumbs up no tests, what are your thoughts here?

Reply:
Oops. This function is annoying to test since calls DB.  I think sufficient to unit test normalising and matching

Reply:
Fine so we can get this out, but it should be not too complicated to just mock the call to other functions, so we would be mostly testing whether this is configured properly (like, it uses JaroWinkler, the 2 rules, and the similarity).